examples/rbcd_attack.py

```diff
@@ -0,0 +1,187 @@
+import functools
+import os.path as osp
+from typing import Optional
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+
+import torch_geometric.transforms as T
+from torch_geometric.datasets import Planetoid
+from torch_geometric.nn import GATConv, GCNConv, GRBCDAttack, PRBCDAttack
+from torch_geometric.nn.conv.gcn_conv import gcn_norm
+from torch_geometric.utils import softmax
+
+dataset = 'Cora'
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'Planetoid')
+# IMPORTANT: Edge weights are being ignored later and most adjacency
+# preprocessings should be part of the model (part of backpropagation)
+dataset = Planetoid(path, dataset, transform=T.NormalizeFeatures())
+data = dataset[0]
+
+
+class GCN(torch.nn.Module):
+    """GCN that normalizes adjacency matrix once for all layers (no cache)."""
+    def __init__(self, hidden_dim: int = 16):
+        super().__init__()
+        gcn_conv = functools.partial(
+            GCNConv,
+            cached=False,  # Important to backpropagate to the adj. matrix
+            add_self_loops=False,  # We add them in `self.gcn_norm`
+            normalize=False  # It is better to normalize once
+        )
```

Comment:
This is interesting. Would prefer to just instantiate via `GCNConv(..., normalize=False)` since I feel this is easier to understand.